lib/logstash/inputs/rabbitmq/march_hare.rb

```diff
@@ -112,6 +112,9 @@ def consume
       @consumer = @q.build_consumer(:block => true) do |metadata, data|
         @codec.decode(data) do |event|
           decorate(event)
+	  metadata.headers.each do |header, value|
+            event[header] = value.respond_to?('toString') ? value.toString : value 
+         end if @include_headers
```

Comment:
By `List ` you meant an array?
Since they aren't a special class, I would probably just take them directly and don't do any conversion. Also I would use a oneliner block for it.

```ruby
metadata.headers.each { |header, value| event[header] = value } if @headers
```